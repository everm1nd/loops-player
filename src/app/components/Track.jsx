import React from 'react';
import Clip from 'components/Clip'

const Track = ({ name }) => <div className="track">
  <h4 className="title">{ name }</h4>
  <Clip key={1}/>
  <Clip key={2}/>
</div>

export default Track;
