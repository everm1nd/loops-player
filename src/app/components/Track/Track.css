.track {
  display: inline-block;
  margin-right: 2px;
}

.track .title {
  margin-bottom: 12px;
  color: #777;
  text-align: center;
}
