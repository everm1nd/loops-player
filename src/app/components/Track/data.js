export default {
  tracks: [
    {
      name: "Track 1",
      clips: [
        { name: "Funky Clip 1"},
        { name: "Funky Clip 2"}
      ]
    },
    {
      name: "Track 2",
      clips: [
        { name: "Soft Clip 1" },
        { name: "Soft Clip 2" }
      ]
    }
  ]
}
