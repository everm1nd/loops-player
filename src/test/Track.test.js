import React from 'react';
import { shallow } from 'enzyme';
import Track from 'components/Track';
import tracksData from "appData";

describe("Track", () => {
  const clips = [
    { id: 0, name: "Funky Clip 1", url: 'funk.mp3', duration: '2m' },
    { id: 1, name: "Funky Clip 2", url: 'funk2.mp3', duration: '2m' }
  ];
  const wrapper = shallow(<Track id={0} name="Track 1" clips={clips} onClick={jest.fn()}/>);

  it('renders a track', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders name", () => {
    expect(wrapper.find(".title").text()).toEqual('Track 1')
  })

  it("renders clips for this track", () => {
    expect(wrapper.find('Clip').length).toBe(2)
  });
})
