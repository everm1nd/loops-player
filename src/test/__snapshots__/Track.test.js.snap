// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track renders a track 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Track
    clips={
        Array [
            Object {
              "id": 0,
              "name": "Funky Clip 1",
              "url": "funk.mp3",
            },
            Object {
              "id": 1,
              "name": "Funky Clip 2",
              "url": "funk2.mp3",
            },
          ]
    }
    color="deeppink"
    id={0}
    name="Track 1"
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <h4
          className="title"
>
          Track 1
</h4>,
        Array [
          <Clip
            id={0}
            name="Funky Clip 1"
            onClick={[Function]}
            playbackState="stopped"
            url="funk.mp3"
/>,
          <Clip
            id={1}
            name="Funky Clip 2"
            onClick={[Function]}
            playbackState="stopped"
            url="funk2.mp3"
/>,
        ],
      ],
      "className": "track",
      "color": "deeppink",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Track 1",
          "className": "title",
        },
        "ref": null,
        "rendered": "Track 1",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": "Funky Clip 1",
        "nodeType": "class",
        "props": Object {
          "id": 0,
          "name": "Funky Clip 1",
          "onClick": [Function],
          "playbackState": "stopped",
          "url": "funk.mp3",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Funky Clip 2",
        "nodeType": "class",
        "props": Object {
          "id": 1,
          "name": "Funky Clip 2",
          "onClick": [Function],
          "playbackState": "stopped",
          "url": "funk2.mp3",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h4
            className="title"
>
            Track 1
</h4>,
          Array [
            <Clip
              id={0}
              name="Funky Clip 1"
              onClick={[Function]}
              playbackState="stopped"
              url="funk.mp3"
/>,
            <Clip
              id={1}
              name="Funky Clip 2"
              onClick={[Function]}
              playbackState="stopped"
              url="funk2.mp3"
/>,
          ],
        ],
        "className": "track",
        "color": "deeppink",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Track 1",
            "className": "title",
          },
          "ref": null,
          "rendered": "Track 1",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": "Funky Clip 1",
          "nodeType": "class",
          "props": Object {
            "id": 0,
            "name": "Funky Clip 1",
            "onClick": [Function],
            "playbackState": "stopped",
            "url": "funk.mp3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Funky Clip 2",
          "nodeType": "class",
          "props": Object {
            "id": 1,
            "name": "Funky Clip 2",
            "onClick": [Function],
            "playbackState": "stopped",
            "url": "funk2.mp3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
